sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py

```diff
@@ -30,21 +32,26 @@ class EventProcessor(EventProcessorMixin):  # pylint:disable=too-many-instance-a
 
     """
     def __init__(
-            self, eventhub_client, consumer_group: str,
-            event_handler: Callable[[PartitionContext, EventData], None],
+            self,
+            eventhub_client: 'EventHubConsumerClient',
+            consumer_group: str,
+            event_handler: Callable[[PartitionContext, EventData], Awaitable[None]],
             *,
-            partition_id: str = None,
-            checkpoint_store: CheckpointStore = None,
-            initial_event_position="-1",
-            initial_event_position_inclusive=False,
+            partition_id: Optional[str] = None,
+            checkpoint_store: Optional[CheckpointStore] = None,
+            initial_event_position: Union[str, int, 'datetime', Dict[str, Any]] = "-1",
+            initial_event_position_inclusive: Union[bool, Dict[str, bool]] = False,
             load_balancing_interval: float = 10.0,
-            owner_level=None, prefetch=None, track_last_enqueued_event_properties=False,
-            error_handler,
-            partition_initialize_handler,
-            partition_close_handler
+            owner_level: Optional[int] = None,
+            prefetch: Optional[int] = None,
+            track_last_enqueued_event_properties: bool = False,
+            error_handler: Optional[Callable[[PartitionContext, Exception], Awaitable[None]]] = None,
+            partition_initialize_handler: Optional[Callable[[PartitionContext], Awaitable[None]]] = None,
+            partition_close_handler: Optional[Callable[[PartitionContext, CloseReason], Awaitable[None]]] = None,
+            loop: Optional[asyncio.AbstractEventLoop] = None
     ):
         self._consumer_group = consumer_group
-        self._eventhub_client = eventhub_client
+        self._eventhub_client = eventhub_client  # type: ignore
```

Comment:
Why? What do you get otherwise?

Reply:
Removed. Doesn't seem to have an impact now. I guess other code changes make this void.